torch/testing/_internal/common_methods_invocations.py

```diff
@@ -12406,6 +12406,8 @@ def error_inputs_mean(op_info, device, **kwargs):
                             device_type='mps', dtypes=[torch.float32]),
                DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit',
                             device_type='mps', dtypes=[torch.float32]),
+               # Could not run 'aten::linalg_eigh' with arguments from the 'AutogradMeta' backend.
+               DecorateInfo(unittest.expectedFailure, 'TestFakeTensorNonErroring', 'test_fake'),
```

Comment:
No clue why I'm seeing this error in this test and in the `test_meta.py`. Any idea @albanD ?

Reply:
There were moves to make Meta a backend bit
cc @bdhirsh do you know what this error is about?

Reply:
it probably has to do with Meta being moved to the backend bit, but that error is pretty strange. Let me see if I can debug locally.

Reply:
Ah, found it....

Diagnosis: we have a meta function for `linalg_eigh` written directly in python, that gets registered to the Meta dispatch key: https://github.com/pytorch/pytorch/blob/bc82a5f79ca056c0f9e5b72f41c07a3387812ac6/torch/_meta_registrations.py#L116.

I think we should leave the python function around in case its useful, but you can remove the `@torch.library.impl(meta_lib, "linalg_eigh")` line that registers it to the dispatcher (since this PR will give us meta tensor support through structured kernels, instead).

Why did that break stuff? `linalg_eigh` is now a `CompositeImplicitAutograd` kernel, which basically means that we should be getting support for it for all backends (including meta) through its decomposition. By registering a meta kernel to it directly, that basically tells the dispatcher "hey, now that I have a direct kernel for the meta backend, then don't run the decomposition anymore when you're using the meta backend

Reply:
I'm also curious what you both think about the naming - we now have a private `_linalg_eigh` op that implements the behavior of the 4 public ops.

Consolidating the implementation into one op seems great, although backends will now need to support a private operator in order to support better linalg coverage (this is the case for a couple other linalg ops too).

I know there have been some conversations around "decomposing into private / unstable API's" being kinda annoying for backends. What do you think about that w.r.t. to the private linalg ops? Are they just more likely to change over time, so making them private is safer?

Reply:
All this are very good points. I had a discussion about this with @mruberry the other day. What we thought is that, for now, the prim operations can simply be those from the public API. Going forward, we may want to have a lower level operation to be the prim with something similar to a LAPACK API, and then have `svd`, `eigh` and so on dispatch to these in Python.

In any case, following the current structure:
- Makes implementing operations for backends fairly easily. If they have the LAPACK-like operation, they can always write two helper functions that call into it less than 10 LOC
- Having these operations written as structured kernels makes prims easy to write, and will make moving to a LAPACK-style API later on rather easy.